packages/block-renderer/src/hooks/use-pattern-inline-css.tsx

```diff
@@ -0,0 +1,53 @@
+import { shuffle } from '@automattic/js-utils';
+import { useMemo } from 'react';
+
+// Shuffle order of blog posts to avoid repetition in previews
+const shufflePostOrder = ( blogPostCount: number ) =>
+	shuffle( [ ...Array( blogPostCount ).keys() ] );
+
+// Memoize order per pattern to show same order in previews
+const inlineCssByPatternId: { [ key: string ]: string } = {};
+
+// Memoize last order
+let lastPostOrder: number[] = [];
+
+const usePatternInlineCss = ( patternId: string, patternHtml: string, isNewSite: boolean ) => {
+	return useMemo( () => {
```

Comment:
I think as discussed `useMemo` here is not being used as we're rendering different components, correct? So we can remove it, and we already are using custom map anyway.